### ISSUE TITLE
Required Vars Are Seen as Unassigned

### ISSUE THREAD

**Issue by ssanbeg** [terraform-aws-modules/terraform-aws-rds#166]
Hello. I am seeing what follows during the **terraform plan** phase via a Makefile:

terraform init --input=false
Initializing modules...

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. --snip --
terraform validate
Success! The configuration is valid.

tflint --deep --aws-region xxxxxx

Error: Unassigned variable

The input variable "allocated_storage" has not been assigned a value. This is
a bug in Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "engine" has not been assigned a value. This is a bug in
Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "instance_class" has not been assigned a value. This is a
bug in Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "engine_version" has not been assigned a value. This is a
bug in Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "identifier" has not been assigned a value. This is a bug
in Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "password" has not been assigned a value. This is a bug in
Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "backup_window" has not been assigned a value. This is a
bug in Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "username" has not been assigned a value. This is a bug in
Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "port" has not been assigned a value. This is a bug in
Terraform; please report it in a GitHub issue.

Error: Unassigned variable

The input variable "maintenance_window" has not been assigned a value. This is
a bug in Terraform; please report it in a GitHub issue.

----
```
module "rds" {
  source                                = "terraform-aws-modules/rds/aws"
  version                               = "2.5.0"

  # 11 variables are required:
  allocated_storage                     = "100"
  backup_window                         = "11:03-11:43"
  engine                                = "MySQL"
  engine_version                        = "8.0.16"
  final_snapshot_identifier             = "${module.appname_dev_rds_instance_label.id}-final"
  identifier                            = module.appname_dev_rds_instance_label.id
  # All available versions: http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_MySQL.html#MySQL.Concepts.VersionMgmt
  instance_class                        = "db.t3.medium"
  maintenance_window                    = "tue:12:11-tue:13:11"
  password                              = data.aws_ssm_parameter.dbpasswd.value
  port                                  = "3306"
  username                              = data.aws_ssm_parameter.dbuser.value

```
----
terraform --version
Terraform v0.12.9
+ provider.aws v2.29.0
+ provider.null v2.1.2

----

Similar to the above this has worked with your vpc and security-group modules just fine. The rds module is acting badly, as you see.

**Comment by tiagocborg** [comment]
I think you will have a better luck reporting this on the terraform issue tracker, since this is a Terraform bug.

**Comment by antonbabenko** [comment]
There was a bug in earlier version of Terraform 0.12 (0.12.10 and older).

It has been fixed in 0.12.12, see [CHANGELOG for Terraform](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md#01212-october-18-2019) + https://github.com/hashicorp/terraform/issues/21659 + https://github.com/hashicorp/terraform/issues/23115 for more details.
